lib/demos/reference/motion_demo_shared_x_axis_transition.dart

```diff
@@ -0,0 +1,239 @@
+                duration: const Duration(milliseconds: 300),
+                reverse: !_isLoggedIn,
+                transitionBuilder: (
+                  child,
+                  animation,
+                  secondaryAnimation,
+                ) {
+                  return SharedAxisTransition(
+                    child: child,
+                    animation: animation,
+                    secondaryAnimation: secondaryAnimation,
+                    transitionType: SharedAxisTransitionType.horizontal,
+                  );
+                },
+                child: _isLoggedIn ? const _CoursePage() : const _SignInPage(),
+              ),
+            ),
+            Padding(
+              padding: const EdgeInsets.symmetric(horizontal: 15),
+              child: Row(
+                mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                children: [
+                  FlatButton(
+                    onPressed: _isLoggedIn ? _toggleLoginStatus : null,
+                    child: Text(localizations.demoSharedXAxisBackButtonText),
+                  ),
+                  RaisedButton(
+                    onPressed: _isLoggedIn ? null : _toggleLoginStatus,
+                    child: Text(localizations.demoSharedXAxisNextButtonText),
+                  ),
+                ],
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class _CoursePage extends StatelessWidget {
+  const _CoursePage();
+
+  @override
+  Widget build(BuildContext context) {
+    final localizations = GalleryLocalizations.of(context);
+
+    return ListView(
+      children: [
+        const SizedBox(height: 16),
+        Text(
+          localizations.demoSharedXAxisCoursePageTitle,
+          style: Theme.of(context).textTheme.headline5,
+          textAlign: TextAlign.center,
+        ),
+        const SizedBox(height: 10),
+        Padding(
+          padding: const EdgeInsets.symmetric(horizontal: 10),
+          child: Text(
+            localizations.demoSharedXAxisCoursePageSubtitle,
+            style: const TextStyle(
+              fontSize: 12,
+              color: Colors.grey,
+            ),
+            textAlign: TextAlign.center,
+          ),
+        ),
+        _CourseSwitch(
+            course: localizations.demoSharedXAxisArtsAndCraftsCourseTitle),
+        _CourseSwitch(course: localizations.demoSharedXAxisBusinessCourseTitle),
+        _CourseSwitch(
+            course: localizations.demoSharedXAxisIllustrationCourseTitle),
+        _CourseSwitch(course: localizations.demoSharedXAxisDesignCourseTitle),
+        _CourseSwitch(course: localizations.demoSharedXAxisCulinaryCourseTitle),
+      ],
+    );
+  }
+}
+
+class _CourseSwitch extends StatefulWidget {
+  const _CourseSwitch({
+    this.course,
+  });
+
+  final String course;
+
+  @override
+  _CourseSwitchState createState() => _CourseSwitchState();
+}
+
+class _CourseSwitchState extends State<_CourseSwitch> {
+  bool _value = true;
```

Comment:
What is the _value representing for the course? Can you rename it to reflect that (isFoo, isEnrolled, etc)